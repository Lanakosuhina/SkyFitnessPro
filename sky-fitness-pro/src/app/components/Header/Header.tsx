
export default function Header() {
  return (
    <div>
      logo skypro
    </div>
  );
}