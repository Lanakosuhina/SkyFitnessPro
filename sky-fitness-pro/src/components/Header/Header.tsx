"use client";
import Link from "next/link";
import Image from "next/image";
import logoImg from "./logo.svg";
import Button from "../Button/Button";
import SVG from "../SVG/SVG";
import { useEffect, useState } from "react";
import DropDown from "../DropDown/DropDown";
import { User, getAuth} from "firebase/auth";
import { app } from "@/app/firebase";

export default function Header() {
  const [isOpen, setIsOpen] = useState(false);
  const [user, setUser] = useState<User | null>(null);

  const toggleDropdown = () => {
    setIsOpen((prevState) => !prevState);
  };

  useEffect(() => {
    const auth = getAuth(app);
    auth.onAuthStateChanged((user) => {    
      if (user) {
        setUser(user);
      } else {
        setUser(user);
      }
    });
  }, []);

  return (
    <>
      <div className="py-14 flex justify-between mx-auto lg:max-w-[1440px] px-4 lg:px-[140px]">
        <div>
          <Link href={"/"}>
            <div>
              <Image src={logoImg} className="w-[220px] h-[35px]" alt="logo" width={220} height={35} />
            </div>
          </Link>
          <p className="hidden md:block pt-3.5 text-[#585959]">
            Онлайн-тренировки для занятий дома
          </p>
        </div>
        {user ? (
          <>
            <div className="relative">
              <div
                onClick={toggleDropdown}
                className="relative top-0 left-0 flex flex-row justify-end gap-4 md:justify-between items-center w-[170px] h-[50px]"
              >
                <SVG icon="icon-user" className="w-[42px] h-[42px]" />
                <div className="flex items-center">
                  <p className="hidden md:block text-2xl font-roboto-400 pr-[11px]">
                    {user.email}
                  </p>
                  <SVG
                    icon="icon-arrow"
                    className={
                      isOpen
                        ? "w-[14px] h-[9px] rotate-180 cursor-pointer"
                        : "w-[14px] h-[9px] cursor-pointer"
                    }
                  />
                </div>
              </div>
              {isOpen  && <DropDown name={user.email} email={user.email}/>}
            </div>
          </>
        ) : (
          <div className="w-[103px]">
            <Link href="/signin">
              <Button title="Войти" />
            </Link>
          </div>
        )}
      </div>
    </>
  );
}
