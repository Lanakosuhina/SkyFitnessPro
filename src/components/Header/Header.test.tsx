import renderer from "react-test-renderer";
import Header from './Header';

describe('Компонент шапки приложения', () => {

  it('корректно рендериться', () => {
    const component = renderer.create(
      <Header />).toJSON();
    expect(component).toMatchSnapshot();
  });

});