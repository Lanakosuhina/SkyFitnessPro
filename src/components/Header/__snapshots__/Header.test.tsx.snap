// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент шапки приложения корректно рендериться 1`] = `
<div
  className="flex justify-between mx-auto py-10 md:pt-[50px] md:pb-14 lg:max-w-[1440px] px-4 md:px-8 main:px-[140px]"
>
  <div>
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <img
        alt="logo"
        className="w-[220px] h-[35px]"
        data-nimg="1"
        decoding="async"
        height={35}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={220}
      />
    </a>
    <p
      className="font-roboto-400 text-lg hidden md:block pt-3.5 text-[#585959]"
    >
      Онлайн-тренировки для занятий дома
    </p>
  </div>
  <div
    className="w-[83px] md:w-[103px]"
  >
    <a
      href="/signin"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="justify-self-center font-roboto-400 text-lg leading-4 rounded-full w-full h-9 py-2 px-4 md:h-[52px] md:px-5 bg-lime  text-black hover:bg-limeHover active:bg-black active:text-white cursor-custom"
      >
        Войти
      </button>
    </a>
  </div>
</div>
`;
